apps/web/modules/storage/utils.ts

```
@@ -0,0 +1,168 @@
+export const sanitizeFileName = (rawFileName: string): string => {
+  if (!rawFileName) return "";
+
+  // Normalize to NFC to avoid weird Unicode composition differences
+  let name = rawFileName.normalize("NFC");
+
+  // Replace path separators/backslashes with dash
+  name = name.replace(/[\\/]/g, "-");
+
+  // Remove control chars and characters problematic in S3 form fields or URLs
+  // Disallow: # <> : " | ? * ` ' and control whitespace
+  name = name.replace(/[\u0000-\u001F#<>:"|?*`'\n\r\t]/g, "");
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)